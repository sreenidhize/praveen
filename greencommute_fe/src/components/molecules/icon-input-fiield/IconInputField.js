import React from "react";
import WorkOutlineIcon from "@material-ui/icons/WorkOutline";
import { makeStyles } from "@material-ui/core";
import BaseIconInputField from "./BaseIconInputField";

const useStyles = makeStyles((theme) => ({
  text: {
    paddingTop: "8px",
    margin: "0 0.5rem",
    fontColor: theme.palette.secondary.main,
    fontWeight: theme.typography.h5.fontSize,
  },
  icon: {
    paddingTop: "12px",
    paddingLeft: "10px",
  },
}));

const IconInputField = (props) => {
  const { variant } = props;
  const classes = useStyles();

  return (
    <BaseIconInputField
      icon={
        props.icon || (
          <WorkOutlineIcon color={variant} className={classes.icon} />
        )
      }
      {...props}
      classes={classes}
    />
  );
};

IconInputField.defaultProps = {
  variant: "secondary",
};

export default IconInputField;
