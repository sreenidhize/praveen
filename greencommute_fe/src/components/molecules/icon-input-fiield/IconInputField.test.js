import React from "react";
import IconInputField from "./IconInputField";
import { render, screen } from "@testing-library/react";
import WorkOutlineIcon from '@material-ui/icons/WorkOutline';


describe("component renders correctly", () => {
  test("should match snapshot", () => {
    const { asFragment, debug } = render(<IconInputField />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("icon and placeholder should not be empty", () => {
    const { debug, container } = render(
      <IconInputField
        icon={<WorkOutlineIcon />}
      />
    );

    expect(container.getElementsByTagName("input").length).toBe(1);
  });
});
