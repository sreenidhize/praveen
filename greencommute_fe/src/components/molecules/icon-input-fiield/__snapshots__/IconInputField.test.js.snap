// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-icon-2 MuiSvgIcon-colorSecondary"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M14 6V4h-4v2h4zM4 8v11h16V8H4zm16-2c1.11 0 2 .89 2 2v11c0 1.11-.89 2-2 2H4c-1.11 0-2-.89-2-2l.01-11c0-1.11.88-2 1.99-2h4V4c0-1.11.89-2 2-2h4c1.11 0 2 .89 2 2v2h4z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="MuiGrid-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-text-1"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInputBase-colorSecondary MuiInput-colorSecondary MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
